packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -470,10 +471,19 @@ export class RunningSummarizer implements IDisposable {
 			let overrideDelaySeconds: number | undefined;
 			let summaryAttempts = 0;
 			let summaryAttemptsPerPhase = 0;
+			// Reducing the default number of attempts to defaultNumberofSummarizationAttempts.
+			const totalAttempts =
+				loggerToMonitoringContext(this.logger).config.getNumber(
+					"Fluid.Summarizer.Attempts",
+				) ?? defaultNumberSummarizationAttempts;
+			assert(
+				totalAttempts >= 1 && totalAttempts <= attempts.length,
+				"Total number of attempts is invalid ",
+			);
 
 			let lastResult: { message: string; error: any } | undefined;
 
-			for (let summaryAttemptPhase = 0; summaryAttemptPhase < attempts.length; ) {
+			for (let summaryAttemptPhase = 0; summaryAttemptPhase < totalAttempts; ) {
```

Comment:
I am wondering if there is ever a situation where `fullTree: true` causes a summary to happen when everything else fails. If so, we won't ever do it now. It's non-blocking for this PR, it's just something that occured to me.

From the data, it doesn't look like it happens often. From the cases I debugged, fullTree: true wasn't the reason why the 4th attempt passed. It was simply because we waited long enough. Is there some kind of logging we can add for this.

Reply:
we are on the same page, I believe.